lightgbm/src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMBase.scala

```diff
@@ -26,6 +26,19 @@ import scala.util.matching.Regex
 trait LightGBMBase[TrainedModel <: Model[TrainedModel]] extends Estimator[TrainedModel]
   with LightGBMParams with HasFeaturesColSpark with HasLabelColSpark with BasicLogging {
 
+  @volatile var sharedState: SharedSingleton[Option[SharedState]] = SharedSingleton(None)
```

Comment:
the state holder should be immutable but the state inside should be mutable right?

Reply:
done, made sharedState immutable